server/grpc_service.go

```diff
@@ -1884,6 +1961,40 @@ func checkStream(streamCtx context.Context, cancel context.CancelFunc, done chan
 	<-done
 }
 
+func (s *GrpcServer) getGlobalTSOFromTSOServer(ctx context.Context) (pdpb.Timestamp, error) {
+	ok, forwardedHost, err := s.GetServicePrimaryAddr(ctx, "tso")
+	if !ok {
+		return pdpb.Timestamp{}, ErrNotFoundTSOADDR
+	}
+	if err != nil {
+		return pdpb.Timestamp{}, errors.WithStack(err)
+	}
+	client, err := s.getDelegateClient(ctx, forwardedHost)
+	if err != nil {
+		return pdpb.Timestamp{}, err
+	}
+	done := make(chan struct{})
+	ctx, cancel := context.WithTimeout(s.ctx, defaultTSOProxyTimeout)
+	go checkStream(ctx, cancel, done)
+	forwardStream, err := tsopb.NewTSOClient(client).Tso(ctx)
```

Comment:
What if the primary changes?

Reply:
it will change by https://github.com/tikv/pd/blob/5744ddd321d3deb728d47dbec92c6efbe8bd0972/server/grpc_service.go#L1965

Reply:
How about the original stream, will it be closed?

Reply:
original stream? with pdpb proto?

Reply:
If we create a stream to A, after that, B becomes primary. Then, we create a stream to B. What about A now?

Reply:
Its behaviour will be similar with other stream in clientConns

Reply:
Yes, but do we need to consider that?

Reply:
The current number of streams is directly related to the tso server, the current number is not much, I will add a todo first. and do we need to think about the eof issue, do we keep a todo first too?